bot/extensions/wikipedia_cog.py

```diff
@@ -0,0 +1,86 @@
+    def __init__(self, search, result):
+        super().__init__()
+        self.search = search
+        self.result = result
+        self.value = None
+
+    @discord.ui.button(label='1', style=discord.ButtonStyle.primary)
+    async def first_wiki_result(self, interaction: discord.Interaction,
+                                button: discord.ui.Button):
+        if (len(self.result[3]) >= 1):
+            await interaction.response.send_message(interaction.user.mention
+                                                    + " requested:\n"
+                                                    + self.result[3][0])
+            self.value = True
+            self.stop()
+        else:
+            await interaction.response.send_message("Invalid choice.",
+                                                    ephemeral=True)
+
+    @discord.ui.button(label='2', style=discord.ButtonStyle.primary)
+    async def second_wiki_result(self, interaction: discord.Interaction,
+                                 button: discord.ui.Button):
+        if (len(self.result[3]) >= 2):
+            await interaction.response.send_message(interaction.user.mention
+                                                    + " requested:\n"
+                                                    + self.result[3][1])
+            self.value = True
+            self.stop()
+        else:
+            await interaction.response.send_message("Invalid choice.",
+                                                    ephemeral=True)
+
+    @discord.ui.button(label='3', style=discord.ButtonStyle.primary)
+    async def third_wiki_result(self, interaction: discord.Interaction,
+                                button: discord.ui.Button):
+        if (len(self.result[3]) == 3):
+            await interaction.response.send_message(interaction.user.mention
+                                                    + " requested:\n"
+                                                    + self.result[3][2])
+            self.value = True
+            self.stop()
+        else:
+            await interaction.response.send_message("Invalid choice.",
+                                                    ephemeral=True)
+
+
+class Wikipedia(Cog, name="Wikipedia", description="Search on Wikipedia."):
+    def __init__(self, bot):
+        self.bot = bot
+
+    @hybrid_command(
+        description="Searches and displays the first 3 results "
+                    "from Wikipedia.")
+    async def wiki(self, ctx, search: str):
+        result = search_results(search)
+        view = Buttons(search, result)
+        if (len(result[1]) == 0):
+            await ctx.interaction.response.send_message("No result found.",
+                                                        ephemeral=True)
+        else:
+            resultView = ""
+            search_count = 0
+            for result in result[1]:
+                search_count += 1
+                resultView += str(search_count) + ": " + result + "\n"
```

Comment:
I see that you are not using the search_count when it is equal to 0, it could be good then to start the counter at 1 then do you first interact with resultView, then increment your counter by 1. Usually, you want to do something before incrementing your variable, it does not really make sense to start incrementing a variable, before using it (extra work for no reason).